### PR TITLE
Move inversion types to analysis module

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
The options for the inversion were out of sync with what was actually being used in the update algorithms. Moved configuration from _es_update to the configuration, and split it on IES and ES.

**Approach**
_Short description of the approach_

image

Forgive my mac:
image

old behavior:

image

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [x] **When there are user facing changes**: Updated documentation
- [x] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/config/analysis_module.py [hunk 1]
```diff
@@ -69,6 +98,10 @@ class IESSettings(BaseSettings):
     """A good start is max steplength of 0.6, min steplength of 0.3, and decline of 2.5",
     A steplength of 1.0 and one iteration results in ES update"""
 
+    inversion: Annotated[
+        InversionTypeIES,
+        Field(title="Inversion algorithm", description=ies_description),
+    ] = "subspace_exact"
```
Comment (oyvindeide) [inline src/ert/config/analysis_module.py:104]: Not that this is changed from the previous behavior (which was direct).

File: src/ert/config/analysis_module.py [hunk 2]
```diff
@@ -41,7 +37,40 @@ class Config:
...
+
+
+InversionTypeIES = Literal["direct", "subspace_exact", "subspace_projected"]
+ies_description = """
+    The type of inversion used in the algorithm. Every inversion method
+    scales the variables. The options are:
+
+    * `direct`:
+        Solve directly, which involves inverting a matrix
+        of shape (num_parameters, num_parameters).
```
Comment (dafeda) [inline src/ert/config/analysis_module.py:61]:
```suggestion
        of shape (num_observations, num_observations).
```

I have created a PR in `iterative_ensemble_smoother` to update the docstring:

https://github.com/equinor/iterative_ensemble_smoother/pull/207/files

**Comment by andreas-el** [comment]
There seems to be an offset issue with the label `Singular value truncation` and the value box. Would it make sense to align the boxes like the used too? Label far left, box far right side?

**Comment by andreas-el** [comment]
Personally im not a fan of choices being lower_case, but that is okay I guess.

Would you consider replacing the labels `exact' etc with bold font instead?

**Comment by oyvindeide** [comment]
> Personally im not a fan of choices being lower_case, but that is okay I guess.
> 
> Would you consider replacing the labels `exact' etc with bold font instead?

The reason I chose lower case is because they are forwarded to another library like that, so tried to stay as close as possible to the end result. I have no qualms with using upper case (though personally I am more of a fan of lower case 🤣)

However, since I wrote upper case in the documentation, I should probably stick with upper case where it is user facing

**Review by andreas-el — APPROVED** [review]
Thanks for addressing these issues, the UI looks better already. 👍

**Pull request merged**